### DISCUSSION TITLE
Using the volatile variable jiffies in a bpftrace program - getting ERROR: Unknown identifier

### DISCUSSION THREAD

**Discussion by tariromukute** [bpftrace/bpftrace#2633] [Q&A] [answered]
Hi. I am trying to calculate the RTT from an SCTP heartbeat packet on the kprobe `sctp_cmd_transport_on`. To do so I need to use the jiffies variable but I am getting error ERROR: Unknown identifier: 'jiffies'. See my program below.

```
bpftrace -e '
#include <net/sctp/structs.h>
#include <linux/jiffies.h>
kprobe:sctp_cmd_transport_on {
    $hbinfo = (struct sctp_sender_hb_info *)arg3->skb->data;
    $rtt = jiffies - $hbinfo->sent_at;
    @[$rtt] = count();
    }'
```

Is it possible for me to reference the kernel jiffies variable, if so what am I missing, if not, are there any workarounds?

**Comment by chengshuyi** [comment, marked as answer]
eBPF already has the bpf_jiffies64 helper function, but it seems that bpftrace does not support it yet

**Reply by tariromukute** [reply]
Thank you @chengshuyi. Does bcc support it? I am guessing my other option is implementing the above as a bcc tool if it supports bpf_jiffies64.

**Reply by chengshuyi** [reply]
yes, bcc support it.

**Reply by viktormalik** [reply]
I opened a new issue #2644 for adding the `jiffies` builtin in case anyone wants to implement it. Thanks for raising this!
